ivn/core.py

```diff
@@ -512,19 +515,23 @@ def create_interface(self):
                 self.logger_topo.warning("ip link {} exists in namespace {} so not create it.".
                                          format(ifname, self.__namespace))
                 return
+        IP_ROUTE.close()
 
+        MAIN_IPDB = IPDB()
         MAIN_IPDB.create(ifname=ifname,
                          kind="veth" if self.__peer else "dummy",
                          peer=self.__peer).commit()
         with MAIN_IPDB.interfaces[ifname] as veth:
             try:
                 veth.net_ns_fd = self.__namespace
             except netlink.exceptions.NetlinkError as e:
+                MAIN_IPDB.release()
                 if e.code == 17:  # "File exists"
                     pass
                 else:
                     raise e
 
+        MAIN_IPDB.release()
```

Comment:
used clause "finally" to combine line 528 and 534

Reply:
There's failure reported in pyroute2 lib if we do this. Haven't figure out why it happens. Thanks for merging.